tests/utils.spec.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @vitest-environment node
+ */
+
+import { describe, expect, it } from 'vitest'
+import { mergeDeep } from '../src/utils'
+
+describe('utils', () => {
+  it('should be possible to replace a primitive value with another', () => {
+    // ARRANGE
+    const state = {
+      foo: 'bar'
+    }
+
+    // ACT
+    const result = mergeDeep(state, {
+      foo: true
+    })
+
+    // ASSERT
+    expect(result).toStrictEqual({
+      foo: true
+    })
+  })
+
+  it('should be possible to merge a nested object', () => {
+    // ARRANGE
+    const state = {
+      foo: {
+        bar: 'baz'
+      }
+    }
+
+    // ACT
+    const result = mergeDeep(state, {
+      foo: {
+        bar: 'bar',
+        foo: 'foo'
+      }
+    })
+
+    // ASSERT
+    expect(result).toStrictEqual({
+      foo: {
+        bar: 'bar',
+        foo: 'foo'
+      }
+    })
+  })
+
+  it('should be possible to replace an array', () => {
+    // ARRANGE
+    const state = {
+      foo: []
+    }
+
+    // ACT
+    const result = mergeDeep(state, {
+      foo: ['bar', 'baz']
+    })
+
+    // ASSERT
+    expect(result).toStrictEqual({
+      foo: ['bar', 'baz']
+    })
+  })
+
+  it('should be possible to add additional key', () => {
+    // ARRANGE
+    const state = {
+      foo: 'bar'
+    }
+
+    // ACT
+    const result = mergeDeep(state, {
+      baz: 'foo'
+    })
+
+    // ASSERT
+    expect(result).toStrictEqual({
+      foo: 'bar',
+      baz: 'foo'
+    })
+  })
+
+  it('should result in the same value as before merging', () => {
+    // ARRANGE
+    const state = {
+      foo: [],
+      bar: []
+    }
+
+    // ACT
+    const result = mergeDeep(state, {
+      foo: [],
+      bar: []
+    })
+
+    // ASSERT
+    expect(result).toStrictEqual({
+      foo: [],
+      bar: []
+    })
+  })
+})
```

Comment:
I think there should be more tests to check if there are no issues with classes, prototype methods, etc.

Maybe you can add them here, or in the `setData.spec.ts` file.

A test that reproduces #2257 would be especially useful

Reply:
I've removed the `utils.spec.ts` file I've created and added a test case that tests the bug of the issue (#2257) and put that into the `setData.spec.ts` file. That file also contains a tests case for checking prototype inheritance.